### PR TITLE
Fixed the promtail Helm Template - DaemonSet doesn't support replicas and strategy statement 

### PR DESCRIPTION
**Pull request by nullck**
```
helm template . --name loki -f values.yaml > loki.yml
k create -f loki.yml --dry-run=true                                                                                                                                                                                                                        
configmap "loki" created (dry run)
configmap "loki-promtail" created (dry run)
serviceaccount "loki" created (dry run)
serviceaccount "loki-promtail" created (dry run)
clusterrole.rbac.authorization.k8s.io "loki-promtail-clusterrole" created (dry run)
clusterrolebinding.rbac.authorization.k8s.io "loki-promtail-clusterrolebinding" created (dry run)
role.rbac.authorization.k8s.io "loki" created (dry run)
role.rbac.authorization.k8s.io "loki-promtail" created (dry run)
rolebinding.rbac.authorization.k8s.io "loki" created (dry run)
rolebinding.rbac.authorization.k8s.io "loki-promtail" created (dry run)
service "loki" created (dry run)
error: error validating "loki.yml": error validating data: [ValidationError(DaemonSet.spec): unknown field "replicas" in io.k8s.api.extensions.v1beta1.DaemonSetSpec, ValidationError(DaemonSet.spec): unknown field "strategy" in io.k8s.api.extensions.v1beta1.DaemonSetSpec]; if you choose to ignore these errors, turn validation off with --validate=false
```
```
k create -f loki.yml --dry-run=true                                                                                                                                                                                                                        [15:55:38]
configmap "loki" created (dry run)
configmap "loki-promtail" created (dry run)
serviceaccount "loki" created (dry run)
serviceaccount "loki-promtail" created (dry run)
clusterrole.rbac.authorization.k8s.io "loki-promtail-clusterrole" created (dry run)
clusterrolebinding.rbac.authorization.k8s.io "loki-promtail-clusterrolebinding" created (dry run)
role.rbac.authorization.k8s.io "loki" created (dry run)
role.rbac.authorization.k8s.io "loki-promtail" created (dry run)
rolebinding.rbac.authorization.k8s.io "loki" created (dry run)
rolebinding.rbac.authorization.k8s.io "loki-promtail" created (dry run)
service "loki" created (dry run)
error: error validating "loki.yml": error validating data: ValidationError(DaemonSet.spec): unknown field "strategy" in io.k8s.api.extensions.v1beta1.DaemonSetSpec; if you choose to ignore these errors, turn validation off with --validate=false
```

Removing ```replicas``` and ```strategy``` statement from the daemonset.yaml , the kubectl create command works good!

### CODE REVIEW

**Comment by huydinhle** [comment]
:+1:

**Comment by kfox1111** [comment]
daemonsets do support strategy's.

**Comment by jgsqware** [comment]
Intereresting, I didn't had that issue.

which version are you using?

**Comment by nullck** [comment]
I'm using the version v1.10.3.
let me test with others versions here.

**Comment by nullck** [comment]
@jgsqware  now I'm testing with the kubernetes version 1.11.6 and I got the same error with the DaemonSet
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T19:44:19Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:30:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

```
k create -f loki.yml --dry-run=true
configmap/loki created (dry run)
configmap/loki-promtail created (dry run)
serviceaccount/loki created (dry run)
serviceaccount/loki-promtail created (dry run)
clusterrole.rbac.authorization.k8s.io/loki-promtail-clusterrole created (dry run)
clusterrolebinding.rbac.authorization.k8s.io/loki-promtail-clusterrolebinding created (dry run)
role.rbac.authorization.k8s.io/loki created (dry run)
role.rbac.authorization.k8s.io/loki-promtail created (dry run)
rolebinding.rbac.authorization.k8s.io/loki created (dry run)
rolebinding.rbac.authorization.k8s.io/loki-promtail created (dry run)
service/loki created (dry run)
error: error validating "loki.yml": error validating data: ValidationError(DaemonSet.spec): unknown field "strategy" in io.k8s.api.extensions.v1beta1.DaemonSetSpec; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by nullck** [comment]
@kfox1111  sorry, I know that the DaemonSet supports strategy but not in the way we have after to render the helm chart.

this which comes with the original chart:
```
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: loki-promtail
  namespace: loki
  labels:
    app: promtail
    chart: loki-0.0.1
    release: loki
    heritage: Tiller
spec:
  selector:
    matchLabels:
      app: promtail
      release: loki
  strategy:
    type: RollingUpdate
```

in this way, for example, works.

```
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: loki-promtail
  namespace: loki
  labels:
    app: promtail
    chart: loki-0.0.1
    release: loki
    heritage: Tiller
spec:
  selector:
    matchLabels:
      app: promtail
      release: loki
  updateStrategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 10%
```

@jgsqware @daixiang0  I'm trying using the version 1.10.6 and now 1.11.6

**Comment by kfox1111** [comment]
Yeah, so I think we should probably just pass through the whole upgrade strategy from the values file so the user can tweak it as necessary rather then remove it.

**Comment by tomwilkie** [comment]
@davkal is going to unblock, sorry for the really slow progress. @jgsqware would also been keen to here your feedback.

File: production/helm/templates/promtail/daemonset.yaml [hunk 1]
```diff
@@ -12,16 +12,10 @@ metadata:
 {{ toYaml . | indent 4 }}
 {{- end }}
 spec:
-  replicas: {{ .Values.promtail.replicas }}
```
Comment (davkal) [inline production/helm/templates/promtail/daemonset.yaml]: Should also be removed in values file.

File: production/helm/templates/promtail/daemonset.yaml [hunk 2]
```diff
@@ -12,16 +12,10 @@ metadata:
...
   selector:
     matchLabels:
       app: {{ template "promtail.name" . }}
       release: {{ .Release.Name }}
-  strategy:
```
Comment (davkal) [inline production/helm/templates/promtail/daemonset.yaml]: Must use `updateStrategy`, see [docs](https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/#performing-a-rolling-update)

**Comment by davkal** [comment]
Superseded by #326

**Pull request closed without merging**